shared/domains/chelonia/chelonia.js

```diff
@@ -91,6 +129,13 @@ sbp('sbp/selectors/register', {
       return stack
     }
   },
+  'chelonia/with-env': async function (contractID: string, env: Object, sbpInvocation: Array<*>) {
+    const savedEnv = this.env
+    this.env = env
+    const res = await sbp('okTurtles.eventQueue/queueEvent', `chelonia/env/${contractID}`, sbpInvocation)
+    this.env = savedEnv
```

Comment:
This looks like the original `env` won't be restored if the `await` statement throws